cmake build out of source broken if source directory not writable

cmake build should work even if the source directory is not writable.  However I am getting an error:
CMake Error: Could not open file for write in copy operation /glade/u/home/turuncu/UFS/ufs-hafs-app/cime/src/externals/pio2/src/clib/pio_meta.h.tmp
CMake Error: : System Error: Permission denied

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -416,7 +416,5 @@ INSTALL(FILES "${PIO_BINARY_DIR}/libpio.settings"
 #####
 configure_file(
   ${CMAKE_CURRENT_SOURCE_DIR}/src/clib/pio_meta.h.in
-  ${CMAKE_CURRENT_SOURCE_DIR}/src/clib/pio_meta.h @ONLY)
+  ${CMAKE_CURRENT_BINARY_DIR}/src/clib/pio_meta.h @ONLY)
 
-FILE(COPY ${CMAKE_CURRENT_SOURCE_DIR}/src/clib/pio_meta.h
-  DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/src/clib/)

